scripts/amd/gemm/matmul.py

```diff
@@ -185,25 +187,24 @@ def matmul_kernel_splitK(
     # while the accumulator is still in FP32!
     if ACTIVATION == "leaky_relu":
         accumulator = leaky_relu(accumulator)
-    c = accumulator.to(tl.float16)
```

Comment:
I'm curious why we don't want to convert to f16? Is it for the atomic_add?

Reply:
So is this just for splitK > 1? If yes, can we do this conditionally if we use splitk or not?

Reply:
I think I know understand why I don't want to convert to f16 now. 

If I revert this line back, 16,16, 1026 is ok, but not for 32, 32, 1026, see below

✅ Triton and Torch match for size, (M: 16, N: 16, K: 1026)
Max difference location (row, column): tensor([ 0, 15])
Value at maxloc in triton output: -73.25
Value at maxloc in torch ouput: -73.3125

❌ Triton and Torch differ for size, (M: 32, N: 32, K: 1026)
Max difference location (row, column): tensor([ 3, 23])
Value at maxloc in triton output: -71.8125
Value at maxloc in torch ouput: -71.875